### DISCUSSION TITLE
How does MeanAbsolutePercentageError deal with divide by zero errors?

### DISCUSSION THREAD

**Discussion by edzWH** [alan-turing-institute/sktime#7112] [Q&A] [answered]
Hi,

I have certain days in my model which I hard code the value to 0 manually, for instance on a Sunday I know the value will be 0 so I hard code this in after the model has predicted the values. 

If I'm calculating the MAPE on this data set against the actuals, there will be many instances where y_pred will be 0 and the actual will be 0.  Would that record be ignored from the overall mean calculation? Or would it contribute towards it as a 0.0 error? 

Should I just omit all the records I know will be 0 prior to calculating the MAPE? Or is this accounted for in some way?

I know MAPE isn't the ideal metric when dealing with 0 values but it is metric I am required to present.

Thanks.

**Comment by fkiraly** [comment, marked as answer]
* the MAPE will not deal explicitly with 0s, just return nan if divisions by zero are encountered - as you would expect from the mathematical object as well
* you can set `symmetric=True` which will produce a symmetrized version of the MAPE, although that will also produce nans if both true and predicted values are zero. This is also what you would expect from the mathematical object (as there is no unique limit to the 0-0 case).

There are various heuristic ways to ensure the 0/0 case is defined, one is adding offsets overall, or when values get too close to 0.

If you have any modified version in mind, we could easily add this to the formula.

**Reply by edzWH** [reply]
Thank you
